sopel/lifecycle.py

```diff
@@ -0,0 +1,180 @@
+"""Deprecation module for Sopel developers and Plugin authors.
```

Comment:
I'm not a fan of German noun style, lol.

```suggestion
"""Deprecation module for Sopel developers and plugin authors.
```